SpringCrud/src/main/java/com/theironyard/controllers/SpringCrudController.java

```diff
@@ -0,0 +1,118 @@
+import javax.servlet.http.HttpSession;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by EddyJ on 7/21/16.
+ */
+@Controller
+public class SpringCrudController {
+
+    @Autowired
+    UserRepository users;
+
+    @Autowired
+    RestaurantRepository restaurants;
+
+    public static final String SESSION_USERNAME = "userName";
+
+    public User getUserFromSession(HttpSession session) {
+        String userName = (String) session.getAttribute(SESSION_USERNAME);
+        User user = users.findFirstByName(userName);
+        return user;
+    }
+
+    @RequestMapping(path = "/", method = RequestMethod.GET)
+    public String home(Model model, HttpSession session){
+        User user = getUserFromSession(session);
+        if(user != null){
+            model.addAttribute("user", user);
+        }
+        List<Restaurant> userRestaurantList = new ArrayList<>();
+        List<Restaurant> otherRestaurantList = new ArrayList<>();
+        for(Restaurant restaurant : restaurants.findAll()){
+            if(restaurant.getUser().equals(user)){
+                userRestaurantList.add(restaurant);
+            }
+            else{
+                otherRestaurantList.add(restaurant);
+            }
+        }
+        model.addAttribute("userRestaurantList", userRestaurantList);
+        model.addAttribute("otherRestaurantList", otherRestaurantList);
+        return "home";
+    }
+
+    @RequestMapping(path = "/login", method = RequestMethod.POST)
+    public String login(HttpSession session, String userName, String password) throws PasswordStorage.CannotPerformOperationException, PasswordStorage.InvalidHashException {
+        User user = users.findFirstByName(userName);
+        if(user == null){
+            user = new User(userName, PasswordStorage.createHash(password));
+            users.save(user);
+        }
+        else if (!PasswordStorage.verifyPassword(password, user.getPassword())){
+            return "home";
+        }
+        session.setAttribute(SESSION_USERNAME, userName);
+        return "redirect:/";
+    }
+
+    @RequestMapping(path = "/logout", method = RequestMethod.POST)
+    public String logout(HttpSession session){
+        session.invalidate();
+        return "redirect:/";
+    }
+
+    @RequestMapping(path = "/create-restaurant", method = RequestMethod.POST)
+    public String createRestaurant(HttpSession session, String restaurantName, String menuType, String restaurantAddress, int restaurantRating){
+        User user = getUserFromSession(session);
+        if(user == null){
+            return "/";
+        }
+        Restaurant restaurant = new Restaurant(restaurantName, menuType, restaurantAddress, restaurantRating, user);
+        restaurants.save(restaurant);
+        return "redirect:/";
+    }
+
+    @RequestMapping(path = "/edit", method = RequestMethod.GET)
+    public String edit(Model model, int id, HttpServletRequest request){
+        String idString = request.getParameter("id");
```

Comment:
This is not needed.  You are already pulling id in as an argument on the line above.  The line below just represents you reassigning what you already have.  I assume this code was copied from your last homework.